woocommerce/woocommerce
# [ci-jobs] Add support for github event and include HPOS disabled e2e tests in ci

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/46925
Closes https://github.com/woocommerce/woocommerce/issues/46924

Adds support for Github events in ci-jobs, so we can only run jobs for specific events, like only for `push`.

- Added a new option in job configuration: `events` as an array of strings, which should contain a list of events for which the job should run. If no events list is defined or the list is empty the job will run for any event.
- Added a new argument for the `ci-jobs` cli: `--event` or `-e`. If no argument is passed the tool will consider all jobs and do not filter by event.
- Moved the HPOS disabled e2e tests into ci.yml by configuring a new project with `events: [push]` configuration. This should only run for `push` events and not run for `pull_request`
- Deprecated 2 workflows for HPOS disabled e2e tests: one for push to trunk and one for daily runs (did not even know this existed until now).
- Improved the HPOS disabled env setup. Because you cannot set environment variables in the job's run command I had to update the `env:test:non-hpos` command and the test env setup script to disable the HPOS feature. Also added a print for the HPOS value in the global setup to easily spot if the expected value is set.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Run pnpm utils ci-jobs with different arguments and check the output.
1. `pnpm utils ci-jobs --event push` should return all configured jobs, including the HPOS disabled ones (5 e2e and 1 API)
2. `pnpm utils ci-jobs --event pull_request` should return all configured jobs without the 5 shards of "Core e2e tests - HPOS disabled" and without the `Core API tests - HPOS disabled`
3. `pnpm utils ci-jobs` should return all configured jobs including the HPOS disabled ones.

- Check CI, the [Build Matrix step](https://github.com/woocommerce/woocommerce/actions/runs/8834156401/job/24255350060?pr=46922#step:4:22) should call `pnpm utils ci-jobs` with the right event argument.
- Tested in CI that HPOS tests having the `pull_request` event configured were triggered: https://github.com/woocommerce/woocommerce/actions/runs/8838048787?pr=46922
- Tested with e2e set to `push` event only (default value) and they were not triggered: https://github.com/woocommerce/woocommerce/actions/runs/8838873869/job/24271155448?pr=46922
